### ISSUE TITLE
Keras -Considering only certain values in Custom Loss Function

### ISSUE THREAD

**Issue by fabridigua** [keras-team/keras#9686]
Hi, this is my problem:
I am building a **CNN**, with an input vector with shape (4000,500,8) but for every 4000 objects there can be less than 500 array of "real" features, the remain features are setted all with arrays of all zeros elements.

`input_layer = layers.Input(shape=(500,8))`
`[...]`
`model = Model(inputs=input_layer, outputs=output_layer)`
`model.compile(optimizer='rmsprop',loss=myCategoricalCrossentropy,metrics=['accuracy'])`
`history = model.fit( [...] ,callbacks=[MyCallback( [...] )])`

Now CNN works ... but analyzing the accuracy of each sample, I saw that for the zero features  ... accuracy is 0.99! As you can imagine this fact makes the result false.
I created a _callback_ that calculates true accuracy .. and works ..

`class MyCallback(Callback):`
`      def __init__(self, train_data, val_data):`
`               [ ... ] `
`      def on_epoch_end(self, epoch, logs={}):` 
`               [ ... ] `
Now I would like to create a custom **loss function** to _make the model only consider the prediction of the "real" features_ ... and not  the others!

`def myCategoricalCrossentropy(yt,yp):`
`#TODO: considering only prediction referring to the real features `
`return` K.categorical_crossentropy(yt,yp)`

Can you help me ??
**Thanks** in advance!
